iam/api-client/src/main/java/com/google/iam/snippets/QuickstartV2.java

```diff
@@ -0,0 +1,166 @@
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+public class QuickstartV2 {
+
+  public static void main(String[] args) {
+    // TODO: Replace with your project ID.
+    String projectId = "your-project";
+    // TODO: Replace with the ID of your member in the form "member:user@example.com"
+    String member = "your-member";
+    // The role to be granted.
+    String role = "roles/logging.logWriter";
+
+    // Initializes the Cloud Resource Manager service.
+    CloudResourceManager crmService = null;
+    try {
+      crmService = initializeService();
+    } catch (IOException | GeneralSecurityException e) {
+      System.out.println("Unable to initialize service: \n" + e.toString());
+    }
+
+    // Grants your member the "Log writer" role for your project.
+    addBinding(crmService, projectId, member, role);
+
+    // Get the project's policy and print all members with the "Log Writer" role
+    Policy policy = getPolicy(crmService, projectId);
+    Binding binding = null;
+    List<Binding> bindings = policy.getBindings();
+    for (Binding b : bindings) {
+      if (b.getRole().equals(role)) {
+        binding = b;
+      }
+    }
+    System.out.println("Role: " + binding.getRole());
+    System.out.print("Members: ");
+    for (String m : binding.getMembers()) {
+      System.out.print("[" + m + "] ");
+    }
+    System.out.println();
+
+    // Removes member from the "Log writer" role.
+    removeMember(crmService, projectId, member, role);
+  }
+
+  public static CloudResourceManager initializeService()
+      throws IOException, GeneralSecurityException {
+    // Use the Application Default Credentials strategy for authentication. For more info, see:
+    // https://cloud.google.com/docs/authentication/production#finding_credentials_automatically
+    GoogleCredential credential =
+        GoogleCredential.getApplicationDefault()
+            .createScoped(Collections.singleton(IamScopes.CLOUD_PLATFORM));
+
+    // Creates the Cloud Resource Manager service object.
+    CloudResourceManager service =
+        new CloudResourceManager.Builder(
+                GoogleNetHttpTransport.newTrustedTransport(),
+                JacksonFactory.getDefaultInstance(),
+                credential)
+            .setApplicationName("service-accounts")
+            .build();
+    return service;
+  }
+
+  public static Policy getPolicy(CloudResourceManager crmService, String projectId) {
+    // Gets the project's policy by calling the
+    // Cloud Resource Manager Projects API.
+    Policy policy = null;
+    try {
+      GetIamPolicyRequest request = new GetIamPolicyRequest();
+      policy = crmService.projects().getIamPolicy(projectId, request).execute();
+    } catch (IOException e) {
+      System.out.println("Unable to get policy: \n" + e.toString());
+    }
+    return policy;
+  }
+
+  private static void setPolicy(CloudResourceManager crmService, String projectId, Policy policy) {
```

Comment:
Could you have the public methods up above the private methods that they call? At first it seemed like there were functions here that weren't used by `main` and were just called from tests, until I followed the code around. Having the public methods first would help this.

Reply:
I reordered them, but I'm not sure it's exactly what you were thinking. lmk if you want me to change something still.